analysis/model/src/main/kotlin/de/maibornwolff/codecharta/util/ResourceSearchHelper.kt

```diff
@@ -5,53 +5,62 @@ import java.io.File
+        }
+
+        fun isFileWithOneOrMoreOfEndingsPresent(resourcePath: String, toBeCheckedFileEndings: List<String>): Boolean {
+            val inputFile = getFileFromStringIfExists(resourcePath) ?: return false
+            if (inputFile.isFile && doesInputFileNameEndWithOneOrMoreOfEndings(inputFile, toBeCheckedFileEndings)) {
                 return true
             }
 
-            val searchFile = File(trimmedResourceName)
+            if (!inputFile.isDirectory) {
+                return false
+            }
 
-            return isResourcePresentInDirectory(searchFile, searchToken, searchOperator, maxSearchingDepth, shouldSearchFullDirectory, resourceShouldBeFile)
+            println("Given resource did not end with any of the supplied file endings. " +
+                    "Scanning directory `${inputFile.absolutePath}` if it contains a file with any of the supplied file endings.")
+            return inputFile.walk()
+                    .asSequence()
+                    .filter { it.isFile && doesInputFileNameEndWithOneOrMoreOfEndings(it, toBeCheckedFileEndings) }
+                    .any()
         }
 
-        fun doesStringEndWith(toBeCheckedString: String, searchToken: String): Boolean {
-            return (toBeCheckedString.endsWith(searchToken))
+        private fun getFileFromStringIfExists(inputFilePath: String): File? {
+            if (inputFilePath == "") {
+                return null
+            }
+            val result = File(inputFilePath.trim())
+            if (result.exists()) {
+                return result
+            }
+            return null
         }
 
-        fun doStringsEqual(string1: String, string2: String): Boolean {
-            return (string1 == string2)
+        private fun isInputFileNameSearchToken(inputFile: File, searchToken: String): Boolean {
+            return(inputFile.name == searchToken)
         }
 
-        private fun isResourcePresentInDirectory(searchFile: File, searchToken: String, searchOperator: (String, String) -> Boolean,
-                                                 maxSearchingDepth: Int, shouldSearchFullDirectory: Boolean, resourceShouldBeFile: Boolean): Boolean {
-            var fileSearch = searchFile.walk()
-
-            if (!shouldSearchFullDirectory) {
-                fileSearch = fileSearch.maxDepth(maxSearchingDepth)
-            }
-
-            return if (resourceShouldBeFile) {
-                fileSearch.asSequence()
-                        .filter { it.isFile }
-                        .map { it.name }
-                        .filter { searchOperator(it, searchToken) }
-                        .any()
-            } else {
-                fileSearch.asSequence()
-                        .map { it.name }
-                        .filter { searchOperator(it, searchToken) }
-                        .any()
+        private fun doesInputFileNameEndWithOneOrMoreOfEndings(inputFile: File, fileEndings: List<String>): Boolean {
```

Comment:
I would rather prefer this method signature:
`endsWithAtLeastOne(inputString: String, endings: List<String>)` It is more generic and not related to a file object, thus it has a better reusability.